openvdb_cmd/vdb_tool/include/Parser.h

```diff
@@ -0,0 +1,1041 @@
+        const std::string str = mData.back();
+        mData.pop_back();
+        return str;
+    }
+    void drop() {// y x -- y
+        if (mData.empty()) throw std::invalid_argument("Stack::drop: empty stack");
+        mData.pop_back();
+    }
+    std::string& top() {
+        if (mData.empty()) throw std::invalid_argument("Stack::top: empty stack");
+        return mData.back();
+    }
+    const std::string& peek() const {
+        if (mData.empty()) throw std::invalid_argument("Stack::peak: empty stack");
+        return mData.back();
+    }
+    void dup() {// x -- x x
+        if (mData.empty()) throw std::invalid_argument("Stack::dup: empty stack");
+        mData.push_back(mData.back());
+    }
+    void swap() {// y x -- x y
+        if (mData.size()<2) throw std::invalid_argument("Stack::swap: size<2");
+        const size_t n = mData.size()-1;
+        std::swap(mData[n], mData[n-1]);
+    }
+    void nip() {// y x -- x
+        if (mData.size()<2) throw std::invalid_argument("Stack::nip: size<2");
+        mData.erase(mData.end()-2);
+    }
+    void scrape() {// ... x -- x
+        if (mData.empty()) throw std::invalid_argument("Stack::scrape: empty stack");
+        mData.erase(mData.begin(), mData.end()-1);
+    }
+    void clear() {mData.clear();}
+    void over() {// y x -- y x y
+        if (mData.size()<2) throw std::invalid_argument("Stack::over: size<2");
+        mData.push_back(mData[mData.size()-2]);
+    }
+    void rot() {// z y x -- y x z
+        if (mData.size()<3) throw std::invalid_argument("Stack::rot: size<3");
+        const size_t n = mData.size() - 1;
+        std::swap(mData[n-2], mData[n  ]);
+        std::swap(mData[n-2], mData[n-1]);
+    }
+    void tuck() {// z y x -- x z y
+        if (mData.size()<3) throw std::invalid_argument("Stack::tuck: size<3");
+        const size_t n = mData.size()-1;
+        std::swap(mData[n-2], mData[n]);
+        std::swap(mData[n-1], mData[n]);
+    }
+    void print(std::ostream& os = std::cerr) const {
+        if (mData.empty()) return;
+        os << mData[0];
+        for (size_t i=1; i<mData.size(); ++i) os << "," << mData[i];
+    }
+};// Stack
+
+// ==============================================================================================================
+
+/// @brief   Implements a light-weight stack-oriented programming language (very loosely) inspired by Forth
+/// @details Specifically, it uses Reverse Polish Notation to define operations that are evaluated during
+///          paring of the command-line arguments (options to be precise).
+class Processor
+{
+    struct Instruction {std::string doc; std::function<void()> callback;};// documentation and callback for instruction
+    using Instructions = std::unordered_map<std::string, Instruction>;
+
+    Stack        mCallStack;// processor stack for data and instructions
+    Instructions mInstructions;// map of all supported instructions
+    Memory       mMemory;
+
```

Comment:
As a general comment, the member variables are usually included at the end of the class definition.